src/Nest/CommonAbstractions/Infer/IndexName/IndexNameResolver.cs

```diff
@@ -15,26 +16,28 @@ public IndexNameResolver(IConnectionSettingsValues connectionSettings)
 
 		public string Resolve(IndexName i)
 		{
-			if (i == null) return this.Resolve((Type)null);
-			return i.Name ?? this.Resolve(i.Type);
+			if (i == null || string.IsNullOrEmpty(i.Name))
+				return this.Resolve((Type)null);
+			if (char.IsUpper(i.Name[0]))
+				throw new ResolveException($"Index names cannot begin with an uppercase character: {i.Name}");
```

Comment:
I "think" an index name can not have uppercase anywhere, not just the first character